src/cli/self_update.rs

```diff
@@ -630,7 +630,17 @@ fn install_bins() -> Result<()> {
     // NB: Even on Linux we can't just copy the new binary over the (running)
     // old binary; we must unlink it first.
     if rustup_path.exists() {
-        utils::remove_file("rustup-bin", &rustup_path)?;
+        if cfg!(windows) {
```

Comment:
One thing to note: the WSLv1 environment is constrained by the underlying NTFS semantics, though MS obviously have improved things substantially - just wondering if we might be better off not making this os-conditional.

Reply:
It certainly would not be a bad idea to make it uniform across all the OSes